File: LEETCODE_CPP/66.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int end = digits.size() - 1;
        while (end >= 0) {
            if (digits[end] == 9) {
                digits[end] = 0;
            } else {
                digits[end] += 1;
                return digits;
            }
            end--;
        }
        digits.insert(digits.begin(),1);
        return digits;
    }
};




int main()
{
    Solution problem;
    vector<vector<int>> test_cases = 
    {
        {1,2,3},
        {4,3,2,1},
        {9},
        {9,9},
        {9,9,9},
        {9,0,0,9},
        {8,9,9},
    };

    for (vector<int> test_case : test_cases)
    {
        problem.plusOne(test_case);
        for (int elements : test_case)
        {
            cout << elements << " ";
        }
        cout << endl; 
    }

    return 0;
}

File: LEETCODE_PYTHON/66.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        
        size = len(digits) - 1

        while (size >= 0):
            if digits[size] == 9:
                digits[size] = 0
            else:
                digits[size] += 1
                return digits
            size -= 1

        digits.insert(0,1)
        return digits



def main():
    solution = Solution()

    testCases = [
        [9],
        [9,9],
        [1,0,0],
        [3,2,0],
        [5],
    ]

    for testCase in testCases:
        returnVal = solution.plusOne(testCase)
        for digit in returnVal:
            print(digit,end="")
        print()


main()